web/src/components/pages/kiosk/kiosk.tsx

```diff
@@ -0,0 +1,200 @@
+import { LocalizationProps, Localized, withLocalization } from 'fluent-react';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import Modal from '../../modal/modal';
+import { CardAction, Button, Hr, LabeledInput } from '../../ui/ui';
+import StateTree from '../../../stores/tree';
+import { User } from '../../../stores/user';
+import { KioskProgress } from '../../../stores/kioskprogress';
+import { DownloadIcon } from '../../ui/icons';
+import PrivacyInfo from '../../privacy-info';
+import URLS from '../../../urls';
+import { RouteComponentProps, withRouter } from 'react-router';
+
+interface PropsFromState {
+  user: User.State;
+  kioskprogress: KioskProgress.State;
+}
+
+interface PropsFromDispatch {
+  updateUser: typeof User.actions.update;
+  clearUser: typeof User.actions.clear;
+  updateKiosk: typeof KioskProgress.actions.update;
+}
+
+interface KioskProps
+  extends LocalizationProps,
+    PropsFromState,
+    PropsFromDispatch,
+    RouteComponentProps<any> {}
+
+class KioskPage extends React.Component<KioskProps> {
+  state: User.State = {
```

Comment:
there's some leftover copypasta. i.e. you don't need all that user state

Reply:
Am I ? I thought it was required to display the email field below

Reply:
Oh, and when I was only pulling email, it would complain about missing fields :)

Reply:
My mistake, I can do it